### ISSUE TITLE
AT does not compile with OpenMP=1

### ISSUE THREAD

**Issue by swhite2401** [atcollab/at#592]
The C code including random number (QuantdiffPass, *QuantPass, etc...) generation does not compile anymore if OpenMP is activate below the error:

![image](https://user-images.githubusercontent.com/10014116/230361424-73b3a718-1c64-4783-b6b0-6e079b948d38.png)

It looks like the random generator has to be added to the shared variables. @lfarv do you confirm?

**Comment by swhite2401** [comment]
In fact adding rng to the shared variables allows to compile correctly, I have test if it runs...

**Comment by lfarv** [comment]
The `rng` variable must indeed be added to the shared variables. I then compiles and runs, but there is another problem: the average energy loss from quantum radiation is ~10% too high with OpenMP.

I'm looking at that.

**Comment by lfarv** [comment]
Update: the test is done looking at the energy loss over one turn of S28F, with 1000 particles and `BndMPoleSymplectic4QuantPass` compared to `BndMPoleSymplectic4RadPass`.

The results are in excellent agreement without OpenMP and differ by ~10% with OpenMP

**Comment by lfarv** [comment]
### Results without OpenMP

![histo1](https://user-images.githubusercontent.com/20038121/230598413-90c49606-a6f3-4f11-bd92-9f816fdd75c6.png)
![curve1](https://user-images.githubusercontent.com/20038121/230598424-0fe5cb5b-8c4b-4738-81cc-32ab33e3d262.png)

Here is the notebook used for the test (remove the .txt extension):
[testquantum.ipynb.txt](https://github.com/atcollab/at/files/11178428/testquantum.ipynb.txt)

**Comment by lfarv** [comment]
### Results with OpenMP:
![histo2](https://user-images.githubusercontent.com/20038121/230599039-bf128c53-a43a-4523-802d-93b81a17e5c4.png)
![curve2](https://user-images.githubusercontent.com/20038121/230599054-8e9505ce-fa1a-476c-b13a-241001594c1e.png)

I do not understand what may happen with the random generator… It would be interesting to do the same test with MPI.

For the time being, it seems necessary to disable OpenMP for the 2 passmethods `BndMPoleSymplectic4QuantPass`, `StrMPoleSymplectic4QuantPass` and `QuantDiffPass`. What do you think ?

**Comment by swhite2401** [comment]
Yes I agree, it has to be disabled. I will check that results are correct with MPI
